### DISCUSSION TITLE
How to aggregate costs using `UsageTypeGroups`

### DISCUSSION THREAD

**Discussion by Haegi** [aws/aws-cli#6956] [Q&A] [answered]
Hi all,

I am currently trying to retrieve cost and usage data for my aws account.
I am able to retrieve the data on a very granular level using the parameters below.
I found out that it is possible to group it on a higher level using `UsageTypeGroups` but unfortunately it is not documented on how to use them. Adding a filter `        "And": {
            "Key": "USAGE_TYPE_GROUP",
            "Values": [
                "EC2: Running Hours"
            ]
        }` seems to not work and as far as I see only one dimension is not allowed.

```json
{
    "TimePeriod": {
      "Start": "2022-05-14",
      "End": "2022-05-15"
    },
    "Granularity": "DAILY",
    "Filter": {
      "Dimensions": {
        "Key": "RECORD_TYPE",
        "Values": [
            "Usage"
        ]
      }
    },
    "Metrics": [
        "UnblendedCost",
        "UsageQuantity"
    ],
    "GroupBy": [
      {
        "Key": "SERVICE",
        "Type": "DIMENSION"
      },
      {
        "Key": "USAGE_TYPE",
        "Type": "DIMENSION"
      }
    ]
}
```

**NOTE**: I am calling it using `aws ce get-cost-and-usage --cli-input-json file:///tmp/ce.json`

**Comment by tim-finnigan** [comment, marked as answer]
Hi @Haegi is this the filter you're looking for?

```
 "Filter": {
      "Dimensions": {
        "Key": "USAGE_TYPE_GROUP",
        "Values": [
            "EC2: Running Hours"
        ]
      }
    },
```

**Reply by Haegi** [reply]
Hi @tim-finnigan it seems to work, thanks!
But is it possible to either use a wildcard or group by `USAGE_TYPE_GROUP` for being able to see the costs and usage for all `USAGE_TYPE_GROUPS`?
I know that it is possible to aggregate on service level but then the UsageQuantity is useless.

**Reply by tim-finnigan** [reply]
Glad that works! I'm not sure about wildcard/grouping for `USAGE_TYPE_GROUP` as that is already a "grouping of common usage types." That being said, it might be worth asking on https://repost.aws/ as someone with more insight into CostExplorer may have a suggestion.
